### PR TITLE
Fix bugs in gauge chart

### PR DESCRIPTION
**Pull request by krkshitij**
## Previous Behavior

- Segment sizes are always rendered as percentages
- The needle moves out of the gauge when the chart value is less than the min value or more than the max value
- Negative segment sizes result in the gauge not being semi-circular
- The chart value can be rendered as a fraction or percentage even when the min value is non-zero
- The SVG contains an aria-label attribute, which conflicts with the presentation role

## New Behavior

- Segment sizes are rendered as percentages or ranges, depending on the variant and minValue props
- The needle rotation stays within a range of 0 to 180 degrees at all times
- Negative segment sizes are treated as 0
- The chart value is rendered as a number when the min value is non-zero and no formatter function is provided
- The SVG doesn't have a presentation role

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6ddc1cc914239b1f50cbfab08c549a20b41a201e'>6ddc1cc914239b1f50cbfab08c549a20b41a201e</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChart.Variant.Example.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { DataVizPalette, GaugeChart, GaugeValueFormat } from '@fluentui/react-charting';
+import { DataVizPalette, GaugeChart, GaugeValueFormat, GaugeChartVariant } from '@fluentui/react-charting';
```
Comment (AtishayMsft) [inline packages/react-examples/src/react-charting/GaugeChart/GaugeChart.Variant.Example.tsx:2]: We should rename this example to single segment.

File: packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx [hunk 2]
```diff
@@ -169,66 +173,58 @@ export class GaugeChartBase extends React.Component<IGaugeChartProps, IGaugeChar
...
+                    fill={segment.color}
+                    fillOpacity={this._legendHighlighted(segment.legend) || this._noLegendHighlighted() ? 1 : 0.1}
+                    strokeWidth={this.state.focusedElement === segment.legend ? ARC_PADDING : 0}
+                    className={this._classNames.segment}
+                    {...getAccessibleDataObject(
+                      {
+                        ariaLabel:
+                          this.props.variant === GaugeChartVariant.SingleSegment
+                            ? `${segment.legend}, ${segment.size} out of ${this._maxValue - this._minValue} or ${(
+                                (segment.size / (this._maxValue - this._minValue)) *
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx:198]: I think we discussed not to have percent if starting value is non zero.

File: packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx [hunk 3]
```diff
@@ -169,66 +173,58 @@ export class GaugeChartBase extends React.Component<IGaugeChartProps, IGaugeChar
...
+                    d={arc.d}
+                    fill={segment.color}
+                    fillOpacity={this._legendHighlighted(segment.legend) || this._noLegendHighlighted() ? 1 : 0.1}
+                    strokeWidth={this.state.focusedElement === segment.legend ? ARC_PADDING : 0}
+                    className={this._classNames.segment}
+                    {...getAccessibleDataObject(
+                      {
+                        ariaLabel:
+                          this.props.variant === GaugeChartVariant.SingleSegment
+                            ? `${segment.legend}, ${segment.size} out of ${this._maxValue - this._minValue} or ${(
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx:197]: Similarly here also, narrate out of if min value = 0

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx [hunk 4]
```diff
@@ -47,36 +47,68 @@ export class GaugeChartPage extends React.Component<IComponentDemoPageProps, {}>
...
               <li>
-                Set the <code>minValue</code> prop if the minimum value of the gauge is different than 0. A placeholder
-                segment will be rendered if the <code>maxValue</code> prop is greater than the total size of the
-                segments.
+                <code>minValue</code>: Use this prop if the minimum value of the gauge is different from 0.
               </li>
               <li>
-                To render an additional text below the chartValue, set the <code>sublabel</code> prop.
+                <code>maxValue</code>: Use this prop to render a placeholder segment when the difference between the max
+                and min values is larger than the total size of the segments. If the difference is smaller, the max
```
Comment (AtishayMsft) [inline packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx:74]: rephrase to - when the desired range for the gauge is more than the sum of all segments.
If the maxValue is less than the sum of all segments, this property is ignored.

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx [hunk 5]
```diff
@@ -47,36 +47,68 @@ export class GaugeChartPage extends React.Component<IComponentDemoPageProps, {}>
...
+                value will be adjusted so that the total size of the segments matches the difference.
               </li>
               <li>
-                To hide the minimum and maximum values of the gauge, set the <code>hideMinMax</code> prop.
+                <code>sublabel</code>: Use this prop to render additional text below the chart value.
               </li>
               <li>
-                The chartValue prop is rendered as a percentage by default. Set the <code>chartVaueFormat</code> prop to
-                'fraction' or a formatter function.
+                <code>hideMinMax</code>: Set this prop to true to hide the min and max values of the gauge.
```
Comment (AtishayMsft) [inline packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx:81]: min and max labels of the gauge.

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx [hunk 6]
```diff
@@ -47,36 +47,68 @@ export class GaugeChartPage extends React.Component<IComponentDemoPageProps, {}>
...
+                </ul>
+                Note: If the min value is non-zero and no formatter function is provided, the chart value will be
+                rendered as a number.
+              </li>
+              <li>
+                <code>variant</code>: This prop determines the presentation style of the gauge chart. Set it to one of
+                the following options:
+                <ul>
+                  <li>
+                    <code>GaugeChartVariant.SingleSegment</code>: Displays and announces the segment sizes as
```
Comment (AtishayMsft) [inline packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx:104]: Clarify this so it means - This is meant to represent a single metric value out of a target goal (n/M).

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChart.Variant.Example.tsx (hunk 1 above)
Comment (krkshitij) [reply packages/react-examples/src/react-charting/GaugeChart/GaugeChart.Variant.Example.tsx:2]: done

File: packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx (hunk 2 above)
Comment (krkshitij) [reply packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx:198]: fixed

File: packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx (hunk 3 above)
Comment (krkshitij) [reply packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx:197]: fixed

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx (hunk 4 above)
Comment (krkshitij) [reply packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx]: done

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx (hunk 5 above)
Comment (krkshitij) [reply packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx]: done

File: packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx (hunk 6 above)
Comment (krkshitij) [reply packages/react-examples/src/react-charting/GaugeChart/GaugeChartPage.tsx]: done

File: packages/react-charting/src/components/GaugeChart/GaugeChart.test.tsx [hunk 7]
```diff
@@ -129,7 +129,107 @@ describe('GaugeChart - snapshot testing', () => {
...
+
+  it('should render the chart value returned by the provided formatter function', () => {
+    const { container } = render(
+      <GaugeChart segments={segments} chartValue={25} chartValueFormat={() => '000'} width={252} height={128} />,
+    );
+
+    expect(container.querySelector('[class^="chartValue"]')).toHaveTextContent('000');
+  });
+
+  it(`should render the chart value as a number when the minValue prop is non-zero and
```
[the remark this answers is not in the document]
Comment (krkshitij) [reply packages/react-charting/src/components/GaugeChart/GaugeChart.test.tsx]: updated tests

File: packages/react-charting/docs/TestPlans/GaugeChart/ComponentTests.md [hunk 8]
```diff
@@ -15,17 +15,27 @@
...
+| -------------------------------------------------------------------------- | ------------------------------------------------------------------------------ | --------- |
+| When the length of the chart value exceeds the max width                   | Should truncate the chart value with ellipsis                                  | RTL       |
+| When the chart resizes                                                     | Should update the font size of the chart value                                 | RTL       |
+| When hideTooltip prop is set to true                                       | Should not show a callout                                                      | RTL       |
+| When chartValue prop is less than the min value or more than the max value | Should ensure the needle rotation remains within the range of 0 to 180 degrees | RTL       |
+| When variant prop is set to GaugeChartVariant.SingleSegment                | Should render segment sizes as percentages                                     | RTL       |
```
[the remark this answers is not in the document]
Comment (krkshitij) [reply packages/react-charting/docs/TestPlans/GaugeChart/ComponentTests.md:26]: Included this in new tests

**Comment by AtishayMsft** [comment]
@krkshitij Could you merge latest master to this PR.

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**